piet/src/gradient.rs

```diff
@@ -233,3 +313,48 @@ impl<P: RenderContext> IBrush<P> for LinearGradient {
         Cow::Owned(piet.gradient(gradient).expect("error creating gradient"))
     }
 }
+
+impl<P: RenderContext> IBrush<P> for RadialGradient {
+    fn make_brush<'a>(&'a self, piet: &mut P, bbox: impl FnOnce() -> Rect) -> Cow<'a, P::Brush> {
+        let rect = bbox();
+        let scale_len = match self.scale_mode {
+            ScaleMode::Fill => rect.width().max(rect.height()),
+            ScaleMode::Fit => rect.width().min(rect.height()),
+        };
+
+        let rect =  equalize_sides_preserving_center(rect, scale_len);
+        let center = self.center.resolve(rect);
+        let origin = self.origin.resolve(rect);
+        let origin_offset = origin - center;
+        let radius = self.radius * scale_len;
+
+        let gradient = FixedGradient::Radial(FixedRadialGradient {
+            center,
+            origin_offset,
+            radius,
+            stops: self.stops.clone(),
+        });
+        // Perhaps the make_brush method should be fallible instead of panicking.
+        Cow::Owned(piet.gradient(gradient).expect("error creating gradient"))
+    }
+}
+
+fn equalize_sides_preserving_center(rect: Rect, new_len: f64) -> Rect {
+    let (x, width) = if new_len != rect.width() {
```

Comment:
I think the "if" here doesn't really help - the conditional branch is likely to be slower than just plowing through the math.

Reply:
interesting, didn't even notice that that would've worked.